### ISSUE TITLE
Tuple unpacking in enums is not handled properly

### ISSUE THREAD

**Issue by Gouvernathor** [microsoft/pyright#5458]
## Environment data

-   Language Server version: 2023.7.10
-   OS and version: Win10
-   Python version (& distribution if applicable, e.g. Anaconda): C 3.11.2

## Code Snippet

```python
class Square(enum.IntEnum):
    (
        A1, B1, C1, D1, E1, F1, G1, H1,
        A2, B2, C2, D2, E2, F2, G2, H2,
        A3, B3, C3, D3, E3, F3, G3, H3,
        A4, B4, C4, D4, E4, F4, G4, H4,
        A5, B5, C5, D5, E5, F5, G5, H5,
        A6, B6, C6, D6, E6, F6, G6, H6,
        A7, B7, C7, D7, E7, F7, G7, H7,
        A8, B8, C8, D8, E8, F8, G8, H8,
    ) = range(64)
    W9 = 68435
    def indexes(self):
        """
        returns file, rank
        """
        r, f = divmod(self, 8)
        return f, r

Square.W9.indexes
Square.A7.indexes # error flagged here : type "int" has no member "indexes"
```

## Expected behavior

PyLance should understand that all X0 attributes of the Square class are Square instance.

## Actual behavior

Pylance correctly deduces the type of W9 which was assigned directly, but fails to understand the type of all the other which were tuple-unpacked.

**Comment by rchiodo** [comment]
Transferring to Pyright. This should be a type checker problem.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release of pyright.
